monkey/infection_monkey/monkey.py

```diff
@@ -199,18 +199,19 @@ def _setup(self):
     def _build_master(self):
         local_network_interfaces = InfectionMonkey._get_local_network_interfaces()
 
-        _event_queue = PyPubSubEventQueue(Publisher())
-        _event_queue.subscribe_type(
-            CredentialsStolenEvent, add_credentials_from_event_to_propagation_credentials_repository
-        )
-
         # TODO control_channel and control_client have same responsibilities, merge them
         control_channel = ControlChannel(
             self._control_client.server_address, GUID, self._control_client.proxies
         )
         credentials_store = AggregatingPropagationCredentialsRepository(control_channel)
 
-        puppet = self._build_puppet(credentials_store)
+        event_queue = PyPubSubEventQueue(Publisher())
+        event_queue.subscribe_type(
+            CredentialsStolenEvent,
+            add_credentials_from_event_to_propagation_credentials_repository(credentials_store),
+        )
```

Comment:
We're going to be having a lot more of these as we add event types. Let's move this to a separate function called `subscribe_events`, or something of that sort, for readability.